biorxiv/corpora_comparison/biorxiv_pmc_figure_generator.py

```diff
@@ -0,0 +1,165 @@
+        lambda row: (
+            f"<b>{row.lemma}</b><br />bioRxiv:{row.biorxiv_count:,}"
+            f"<br />PMC:{row.pmc_count:,}<br />"
+            f"Odds Ratio:{row.odds_ratio:.2f}"
+        ),
+        axis=1
+    ))
+    
+)
+plot_df.head()
+
+
+# In[5]:
+
+
+fig=go.Figure(
+    go.Treemap(
+        branchvalues="remainder",
+        labels = ["bioRxiv","PMC"] + plot_df.label.tolist(),
+        parents = ["", ""] + plot_df.repository.tolist(),
+        values = [0,0] + plot_df.odds_ratio.tolist(),
+        textinfo = "label",
+        textfont = {'size':17}
+    )
+)
+fig.update_layout(
+    width=1280, 
+    height=720,
+    treemapcolorway=["#a6cee3", "#b2df8a"],
+    title={
+        'text':f"Comparison of Top {plot_df.shape[0]} Words between bioRxiv and Pubmed Central (PMC)",
+        'y':0.9,
+        'x':0.5,
+        'xanchor': 'center',
+        'yanchor': 'top'}
+)
+fig.update_layout(uniformtext=dict(minsize=10))
+fig.show()
+
+
+# In[6]:
+
+
+fig.write_image("output/figures/biorxiv_vs_pubmed_central.svg", scale=2.5)
+fig.write_image("output/figures/biorxiv_vs_pubmed_central.png", scale=2.5)
+
+
+# # Preprint vs Published
+
+# In[7]:
+
+
+preprint_published_comparison = pd.read_csv("output/preprint_to_published_comparison.tsv", sep="\t")
+preprint_published_comparison.head()
+
+
+# In[8]:
+
+
+preprint_published_comparison = (
+    preprint_published_comparison
+    .assign(repository=lambda x: x.odds_ratio.apply(lambda ratio: "preprint" if ratio > 1 else "published"))
+    .sort_values("odds_ratio", ascending=False)
+)
+preprint_published_comparison
+
+
+# In[9]:
+
+
+subset = 20
+plot_df = (
+    preprint_published_comparison.head(subset).append(preprint_published_comparison.tail(subset))
+    .assign(odds_ratio=lambda x: x.odds_ratio.apply(lambda ratio: ratio if ratio > 1 else 1/ratio))
+    .assign(label=lambda x: x.apply(
+        lambda row: (
+            f"<b>{row.lemma}</b><br />Preprint:{row.preprint_count:,}"
+            f"<br />Published:{row.published_count:,}<br />"
+            f"Odds Ratio:{row.odds_ratio:.2f}"
+        ),
+        axis=1
+    ))
+    
+)
+plot_df.head()
+
+
+# In[10]:
+
+
+fig=go.Figure(
+    go.Treemap(
+        branchvalues="remainder",
+        labels = ["preprint","published"] + plot_df.label.tolist(),
+        parents = ["", ""] + plot_df.repository.tolist(),
+        values = [0,0] + plot_df.odds_ratio.tolist(),
+        textinfo = "label",
+        textfont = {
+            'size':15
+        }
+    )
+)
+fig.update_layout(
+    width=1280, 
+    height=720,
+    treemapcolorway=["#a6cee3", "#b2df8a"],
+    title={
+        'text':f"Comparison of Top {plot_df.shape[0]} Words between Preprints and Published Papers",
+        'y':0.9,
+        'x':0.5,
+        'xanchor': 'center',
+        'yanchor': 'top'}
+)
+fig.update_layout(uniformtext=dict(minsize=10))
+fig.show()
+
+
+# In[11]:
+
+
+fig.write_image("output/figures/preprint_published_comparison.svg", scale=2.5)
```

Comment:
Do you want to remove the degree symbol and single letters for this analysis?

Also for preprint vs published, are these preprints that have NOT been published? So is the published a subset of the preprints? I assume not given the strong difference in words but just wanted to check.

Reply:
> Do you want to remove the degree symbol and single letters for this analysis?

I thought about removing the single letters and symbols, but ultimately decided to keep them in the analysis as they pertain to math symbols and measurements. Good to note for future analysis of this dataset!

> Also for preprint vs published, are these preprints that have NOT been published? So is the published a subset of the preprints?

No. These are only the preprints that have gone through the publishing process. The idea here is to see how a research article changes when it has been accepted by a journal.

Reply:
Ah..so does that mean the articles included in the preprint and published should be the same, but the text content might be different and thats what you're comparing?

Reply:
>Ah..so does that mean the articles included in the preprint and published should be the same, but the text content might be different and thats what you're comparing?

Bingo! That is correct. The expectation here is that text content should change as reviewers and editors suggest revisions towards an article.